implementation-details.md

```diff
@@ -284,32 +313,53 @@ The `CrdsFilter`s are partitioned in pull requests between peers selected random
 
   where: 
   * `node_stake` - stake of "our" node, 
-  * `peer_stake[i]` - stake of i-th node from the list.
+  * `peer_stake[i]` - stake of i-th peer from the list.
+
   
-The higher the node's weight, the more pull requests will be associated with it.
+The higher the node's weight, the more pull requests will be sent to it.
 
-Each pull request contains one of the `CrdsFilter`s from the list and a `CrdsValue`, which is usually a `LegacyContactInfo`. It contains details of the node that sent the pull request - a list of its IP addresses, public key and wallclock. 
+Each pull request contains one of the `CrdsFilter`s from the list and a `CrdsValue`, which is a `LegacyContactInfo`. It contains details of the node that sent the pull request - a list of its IP addresses, public key and wallclock. 
```

Comment:
nit: maybe say: "...from the list and the requesting node's `LegacyContactInfo` as a `CrdsValue`..." or something

Reply:
> It contains details of

Instead say: The node's `LegacyContactInfo` contains details ...